### PR TITLE
Hyundai: Ioniq 6 support

### PR DESCRIPTION
**Pull request by sshane**
0x2a4 is missing, there's not a lot of CAM messages. @sunnyhaibin would you like to take this one over? It may require a bit more than a simple fingerprint. The user is on Discord under a thread in the hyundai channel

### CODE REVIEW

File: selfdrive/car/hyundai/values.py [hunk 1]
```diff
@@ -187,6 +188,10 @@ def init_make(self, CP: car.CarParams):
     HyundaiCarInfo("Hyundai Ioniq 5 (without HDA II) 2022-23", "Highway Driving Assist", harness=Harness.hyundai_k),
     HyundaiCarInfo("Hyundai Ioniq 5 (with HDA II) 2022-23", "Highway Driving Assist II", harness=Harness.hyundai_q),
   ],
+  CAR.IONIQ_6: [
+    HyundaiCarInfo("Hyundai Ioniq 6 (without HDA II) 2023", "Highway Driving Assist", harness=Harness.hyundai_k),  # TODO: unknown
+    HyundaiCarInfo("Hyundai Ioniq 6 (with HDA II) 2023", "Highway Driving Assist II", harness=Harness.hyundai_q),  # TODO: unknown
```
Comment (sshane) [inline selfdrive/car/hyundai/values.py:193]:
```suggestion
    HyundaiCarInfo("Hyundai Ioniq 6 (with HDA II) 2023", "Highway Driving Assist II", harness=Harness.hyundai_p),
```

**Comment by adeebshihadeh** [comment]
> 0x2a4 is missing, there's not a lot of CAM messages.

That's used to disable LFA on HDA2 cars. Need to make sure LFA is still blocked while openpilot is running.

**Comment by gomtings** [comment]
An error occurs when annotating create_cam_0x2a4. It needs to be restored or dealt with. The vehicle I checked is Ioniq 5.

**Comment by ntthuy11** [comment]
is this work still active? I am going to buy a Ioniq 6 + comma kit if I6 is supported. Thanks for the work!

**Comment by sshane** [comment]
Yes! We're just lacking testers at the moment.

**Comment by IoniqYeti** [comment]
Hello, I believe that I finally came to the right spot to find support for the ioniq6! I have installed the comma 3 in my ioiniq6. 

I am running a branch of the openpilot called ioniq - 6. I have talked with some people in the discord and they have told me that it is not a fingerprint issue. The comma3 will recognize my device but I'm unable to use any of the features of openpilot.

In this branch, the comma 3 recognizes the car and will not say dashcam mode BUT my dashboard lights up like a Christmas tree. I get 6 errors in total and ALL of my driver assist features are disabled and I cannot engage any of the driving assist features.

If I disable open pilot in the toggles menu, then the car goes to dashcam mode and there are no errors or warnings on the dashboard.

my dongle ID is 5b86a6f0102bb5e9

here is a route ID with the openpilot system running 5b86a6f0102bb5e9|2023-07-03--06-06-41--0

here is a route ID with openpilot system off in toggles (dashcam mode) with me activating hda2 capabilities 5b86a6f0102bb5e9|2023-07-03--12-43-54--15

**Comment by sshane** [comment]
@IoniqYeti Strange, I do not see the camera sending LKAS, only LFA, which might cause certain ECUs to fault if it sees a message it's not expecting. Can you turn on alpha longitudinal in the settings and see if that resolves it? It tries to send LFA to steer in that state

**Comment by IoniqYeti** [comment]
With open pilot on and alpha longitudinal I now get a CAN Bus disconnected:likely faulty cable error. Still unable to activate any LFA or smart cruise control.

**Comment by IoniqYeti** [comment]
I also flipped a bunch of cables one at a time to see if it would be that but it's the same issue. I also sometimes just get a harness relay malfunction check hardware error on comma

**Comment by IoniqYeti** [comment]
route with openpilot + alpha longitudinal = harness relay malfunction initially and turns into can bus disconnected: likely faulty cable error
5b86a6f0102bb5e9|2023-07-05--14-50-22--0

**Comment by sshane** [comment]
@adeebshihadeh it would be much easier to find the message to disable HDA if we rented this car eventually, else I'll have users test branches in the mean time when I get time.

**Comment by sshane** [comment]
community has taken this PR up https://github.com/commaai/openpilot/pull/29486

**Pull request closed without merging**